### DISCUSSION TITLE
How do I make SMPlayer use external subtitles by default?

### DISCUSSION THREAD

**Discussion by MultiKoopa** [smplayer-dev/smplayer#907] [Q&A]
I have, in the same folder, two files: an mkv file, and a .SRT file. both have the exact same name before the file extension.

for some reason, SMPlayer no longer defaults to that external file, even if there are no other subtitles available. 

Under Subtitles, Autoload is set to "all subtitles containing the movie name"

Preferred language for Subtitles is not set.

Windows 11 23H2 latest update (64-bit), SMPlayer version 23.12.0.10207

**Comment by smplayer-dev** [comment]
What version are you using?
What operating system?

**Reply by MultiKoopa** [reply]
whoops sorry about that

Windows 11 23H2 latest update (64-bit), SMPlayer version 23.12.0.10207

**Reply by residentbrick** [reply]
Seeing same problem

Version: 23.12.0 (revision 10207) (64-bit)
Using Qt 5.15.2 (compiled with Qt 5.15.2)
Using MPV 0.36.0

Windows 11 Home 22H2 10/‎22/‎2022 22621.3155 Windows Feature Experience Pack 1000.22684.1000.0

Used SMPLAYER for many years.  When I place MP4 and SRT in same folder using same base name, SMPLAYER always
identified the subtitle file.  Just upgraded to current SMPLAYER and now have to identify the subtitle track for a new
video file.

**Comment by smplayer-dev** [comment]
I can't reproduce this problem.
Can you send the srt file?

**Reply by MultiKoopa** [reply]
It's quite literally any video file that hasn't been opened before, with any subtitle file; it will never select either internal or external subs by default.

**Comment by smplayer-dev** [comment]
I'd like to fix it but I can't reproduce it on Windows 11, with SMPlayer 23.12.0 10207.
Is the option "Select first subtitle available" in preferences -> subtitles enabled?

**Reply by MultiKoopa** [reply]
it wasn't; I turned it on, and... no difference.

**Comment by residentbrick** [comment]
I am running some tests using the latest SMPlayer.  I don't know if this is important, but I noticed a couple little peculiarities.

When I load a "new" MP4 file into SMPlayer (by "new" I mean that it has never been previously loaded so there is no possibility of resuming), and have the corresponding SRT file in the same folder with the same base filename, I have come to expect that the
video file will begin playing with subtitles from the SRT displayed.  That is not what happens, however.  It begins playing WITHOUT subtitles.

I noticed that the "Select subtitles track" icon at the of the SMPlayer toolbar, shows both "None" and the SRT file.  However, neither item has a bullet marking which one is active.  

Now, if I select the SRT file, then the subtitles from the SRT file will display.

There is ALSO an error or warning message which appears for just a few seconds near the top of screen, in a font similar to the 
subtitles, but not in the expected subtitle area.  This message displays 

sid(1) Unknown("01 Alias S01E04 A Broken Heart.srt")

where the double quoted string is the name of the SRT file being used.  

The sid(1) message appears whenever the SRT is selected and then quickly vanishes.  I don't know if this is significant.  But since I never had to select the SRT file before to enable subtitles with SMPlayer, it may well be innocuous.

I also have an older laptop which I rarely use anymore.  It runs Windows 11, but I think it has a version of SMPlayer which I have not
yer updated.  I am going to try the same experiment on that system using a "new" MP4 video, to assure myself that I did not have to set the SRT subtitle track to get subtitles displayed.

**Comment by smplayer-dev** [comment]
Could you send that srt file?
I can't reproduce this problem with the videos and subtitles I have.

**Reply by MultiKoopa** [reply]
I'm telling you it's any srt file with any mkv file.

I downgraded to 23.6.0 and the problem immediately resolved.

**Comment by residentbrick** [comment]
Just as MultiKoopa says, its NOT the SRT file. 

Here are the results of testing on my old laptop.  This laptop is running SMPlayer 22.2.0 (revision 10060)

I open an MP4 video file and its associated SRT file.  They reside in the same folder.  The base file names are the same so there is no
question that the SRT file is associated with the MP4 video file.  The video file and the SRT have never been played on this laptop. 
So there can be no resuming of play by SMPlayer.

SMPlayer begins playing the video AND using the SRT file for subtitles.   This is quite different from the actions with the current SMPlayer which required setting the Subtitle track before subtitles would display.

I checked the setting of the Select subtitles track icon at the top of the SMPlayer screen.  After loading the video and play starting,
the Select subtitles track icon had the SRT file marked with a bullet.  This, also, is different from the current SMPlayer.

The sid(1) Unknown() message appears when I change the state of the Select subtitles track from "None" to the SRT file.  This
is the same as the current SMPlayer.  I don't know what the message means, but it is apparently normal.

MultiKoopa resolved his problem by downgrading SMPlayer.

I'll watch for news of a resolution and will remember to manually set the SRT track in the meantime.

**Comment by smplayer-dev** [comment]
I think it could be caused by the mpv version (0.36) included in that package. However since I can't reproduce this issue I can't be sure. As it works for me it should be either something strange in the srt files or something in the smplayer configuration.

**Reply by residentbrick** [reply]
As I understand it, then, nothing will be done and I should downgrade immediately and begin searching for an alternative to SMPlayer?

**Reply by smplayer-dev** [reply]
You can send me an srt file and/or your smplayer.ini so I can try to find out the problem.

**Reply by residentbrick** [reply]
The SRT file is not important.  The two tests I  made with the current SMPlayer and an older version show that in both cases, SMPlayer recognized the presence of the SRT file and listed it as a valid entry for "Select subtitle track".  In the older version, SMPlayer actually marked it with a bullet indicating that it was selected for subtitlles.  This only happens with the current SMPlayer when resuming play or when selecting the subtitle track a second time.  Since neither option was marked, no subtitles were initially displayed.  

I will try to send the smplayer.ini file.

**Reply by residentbrick** [reply]
Sending the smplayer.ini file again.

[smplayer.zip](https://github.com/smplayer-dev/smplayer/files/14521292/smplayer.zip)

**Comment by raszpl** [comment]
> The SRT file is not important.

you know you could just sent that damn str file two weeks ago to let smplayer-dev check it for himself and move on with debugging :)

**Comment by smplayer-dev** [comment]
I'm also waiting for the smplayer.ini file.

**Reply by residentbrick** [reply]
I sent smplayer.ini file weeks ago.  If you didn't get, I'll go back and find out what happened.

**Reply by Raza-0007** [reply]
Sent how? Through Postal Mail :confused:

All you need to do is to attach the files with your post! 

~Raza

**Reply by residentbrick** [reply]
I sent it HERE HERE HERE HERE HERE where it says "Paste, drop, or click to add files". It was necessary to put the smplayer.ini file into
a ZIP file as sending the INI file was not allowed.  

[smplayer.zip](https://github.com/smplayer-dev/smplayer/files/14521118/smplayer.zip)

What I determined was that the SMPlayer or the underlying application no longer automatically selects the subtitle track as it had in the past. The SRT file is displayed as an available track, but it remains unselected (along with "None") for the initial load of a video and SRT file into the player.  Taking the extra step of manually selecting the SRT file corrects the problem immediately and for future re-loads.

**Comment by smplayer-dev** [comment]
I've just tested you smplayer.ini.
After enabling the option "Select first available subtitle" in preferences -> subtitles, the problem gets fixed for me.

**Reply by residentbrick** [reply]
Thats it.  Problem solved.  Thanks for pointing out the option. Close this baby out.

**Comment by residentbrick** [comment]
I shouldn't have agreed so quickly to closing.  Its still doing it.   Just as MultiKoopa indicated in the first post of this thread., under Subtitles, Autoload is set to "all subtitles containing the movie name". I also have set "Select first available subtitle".  For a video which has not yet been loaded into SMPlayer, and the video and SRT share the same base name, the subtitle are STILL not displayed initially.  Verified that those options are set correctly, but still no subtitles.  Once I click on the Subtitles tool and select the proper subtitle file, subtitles display fine. I am just going to accept that this is a bug which will never be fixed.  The work around is simple.  Just an initial manual set for subtitles on first load.  So, not a big deal.  I'lll live with the bug.

**Reply by MultiKoopa** [reply]
I'll just use the old smplayer version without the bug, I guess.

**Reply by Raza-0007** [reply]
When there is a bug like this that only effects a couple of people, I usually recommend that you reset your smplayer settings to factory defaults. You can do so by deleting the `smplayer.ini` and `playlist.ini` files inside the smplayer configuration folder. Restart smplayer and new default files will be generated for you. You will have to reset all your current settings, but hopefully this bug will be resolved.

~Raza

**Reply by residentbrick** [reply]
Thanks.  I will try that and report back later. I just ran test using a video & SRT file which had never been loaded into SMPlayer.  Results were the same.  The "Select subtitle track" tool displays both None and the SRT filename, but neither are selected even though that is the option set in preferences. Once the SRT file is selected, play of the video is properly subtitled.

I'll repeat the test later with a different, unplayed video and SRT combo after deleting the two ini files.

**Reply by residentbrick** [reply]
Deleted the smplayer.ini and playlist.ini files.  Tested using new unplayed video and SRT.  Results are the same.  The "Select subtitle track" shows the SRT file along with the None option.  Neither are marked/selected.  Once I select the SRT file, subtitles are correctly displayed.
Reloading this video or any previously loaded video and subtitles are correctly displayed.  I'll continue to manually set the subtitle track for any new, unplayed video.  It only takes a second.  Not a big deal. But thanks for help.

**Comment by smplayer-dev** [comment]
Could you send the smplayer log (menu View) after playing a video with this problem?

**Reply by residentbrick** [reply]
Will do.  Just takes a few minutes to download a video and SRT that I can be sure have never been played before.

**Comment by residentbrick** [comment]
Playing an episode from 90s TV show, Alias.  Episode 7.  Color Blind.  Played video for perhaps 20 seconds.  There should have been some subtitles displayed, but nothing appeared.  Paused the video.  Manually set the subtitle track.  The SID(1) message appears, then vanishes. Then the subtitle for the point in the video where it was paused appears.

[smplayer.log.txt](https://github.com/smplayer-dev/smplayer/files/14576306/smplayer.log.txt)

**Reply by residentbrick** [reply]
I only now noticed in Preferences -> General -> Preferred audio and subtitles.  The bottom section "Or choose a track number".  I changed the Subtitle value which had been 0 to 1, just to see what this would do.   When SMPlayer loaded a video not yet seen, there was a very brief message.  It might have been the SID(1) message.  I could not tell.  But the SMPlayer began playing the video and displaying subtitles. On a later resume of the video, subtitles were also displayed, this time without the warning message.  I don't know this was relevant, but I thought I would mention it.

**Comment by smplayer-dev** [comment]
I could reproduce the issue but only once. 
I don't understand why you have this problem every time but I can hardly see it.

Does it make any difference if you add this option in preferences -> advanced -> mplayer/mpv -> options:
`--sid=auto`

![Captura desde 2024-03-13 15-22-55](https://github.com/smplayer-dev/smplayer/assets/10128951/3be59f23-0a0e-4373-a6ec-a05cf0bffc3c)

**Reply by residentbrick** [reply]
I'll try it out and report back.  But if you were able to reproduce it once, that may be it.  It only happens the VERY FIRST TIME a video is loaded into SMPlayer.  On subsequent plays, SMPlayer remembers that subtitles come from an external SRT file and correctly resumes play at the proper point displaying those subtitles.  That is why we love SMPlayer.  VLC Media player was constantly forgetting where it left off.  No such problem with SMPlayer.   It should be noted that I was unable to fool SMPlayer.  I tried copying the video to a new folder, renaming the video and SRT.  SMPlayer saw through my feeble attempts at trickery and resumed perfectly.

**Reply by residentbrick** [reply]
Using --sid=auto worked and saved me from having to manually select the SRT file for the subtitle track.  Is there a way to suppress the annoying SID(1) message when the subtitle track is selected either manually or by --sid=auto.  I don't need to see the SID(1) message on the screen. I put the SRT file in the same folder as the video.  That ought to be sufficient to enable the subtitle display without additional confirmation.  That was how older versions of SMPlayer operated. The SID(1) message does not appear when the video is reloaded or resumed.  So, thats good.

**Reply by Raza-0007** [reply]
@residentbrick,

You can actually delete the contents of the folder `file_settings` inside smplayer configuration folder to make smplayer forget about the files it has played before. You do not need to download a new file every time to test this.

~Raza

**Comment by Raza-0007** [comment]
@residentbrick and @smplayer-dev,

I had completely forgotten but I had helped somebody with this issue before. It is actually not an smplayer issue, but an mpv issue, and it was fixed in mpv version 0.37

[See this discussion](https://github.com/smplayer-dev/smplayer/discussions/873)

So they changed something in mpv 0.36 which caused this issue, and then reverted it back to the mpv 0.35 logic, in mpv 0.37

So upgrading mpv to 0.37 will fix it. I myself have been running mpv 0.37 since December, this is why I don't have this issue.

~Raza

**Reply by MultiKoopa** [reply]
ok, manually updating mpv fixed it. Thank you!

**Reply by residentbrick** [reply]
Thanks, this gives me new hope.  With all the tweaking of SMPlayer preferences, I made things substantially worse.  Now I have to select the subtitle track each time.  Instead of mucking around further, I'm going to wipe SMPlayer and then reinstall and manually update mpv.

**Comment by smplayer-dev** [comment]
The packages for Windows in this build use mpv 0.37. Please test if the problem is solved now:

https://github.com/smplayer-dev/smplayer/releases/tag/v23.12.0.10242
